Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -588,34 +692,91 @@ private void verifyElementSeen(@NonNull String tag, boolean hasSeenTag) throws K
     }
   }
 
+  private void verifyIdNotInMap(Map map, String id) throws KMLParseException {
+    if (map.containsKey(id)) {
+      throw new KMLParseException(
+          "Error at: " + mParser.getPositionDescription() + " ID " + id + " already seen.");
+    }
+  }
+
+  private void mergeStyles() {
```

Comment:
It's not obvious that this method merges shared styles into inline styles. Consider alternate method name as well as a comment describing in more detail.